mava/systems/tf/maddpg/system.py

```diff
@@ -25,26 +25,33 @@
 import sonnet as snt
 from acme import specs as acme_specs
 from acme.tf import utils as tf2_utils
-from acme.utils import counting, loggers
+from acme.utils import loggers
+from dm_env import specs
 
 import mava
 from mava import core
 from mava import specs as mava_specs
-from mava.components.tf.architectures import DecentralisedQValueActorCritic
+from mava.components.tf.architectures import (
+    DecentralisedQValueActorCritic,
+    DecentralisedValueActorCritic,
+)
 from mava.environment_loop import ParallelEnvironmentLoop
 from mava.systems.tf import executors
-from mava.systems.tf import savers as tf2_savers
 from mava.systems.tf.maddpg import builder, training
-from mava.systems.tf.maddpg.execution import MADDPGFeedForwardExecutor
-from mava.utils import lp_utils
+from mava.systems.tf.maddpg.execution import (
+    MADDPGFeedForwardExecutor,
+    sample_new_agent_keys,
+)
+from mava.systems.tf.variable_sources import VariableSource as MavaVariableSource
 from mava.utils.loggers import MavaLogger, logger_utils
+from mava.utils.sort_utils import sort_str_num
 from mava.wrappers import DetailedPerAgentStatistics
 
 
 class MADDPG:
     """MADDPG system."""
 
-    def __init__(
+    def __init__(  # noqa
```

Comment:
Why do you need the `# noqa` here?

Reply:
For some reason, mypy is complaining that the function is too complex. I am guessing it has too many arguments.